crates/matrix-sdk/src/error.rs

```diff
@@ -175,12 +175,15 @@ impl HttpError {
                     }
                     _ => Some(e.status_code),
                 },
-                RumaApiError::Uiaa(_) => None,
                 RumaApiError::Other(e) => Some(e.status_code),
+                RumaApiError::Uiaa(_) => None,
             };
 
             if let Some(status_code) = status_code {
-                if status_code.is_server_error() {
+                // If the status code is 429, this is requesting a retry in HTTP, without the
```

Comment:
MDN tells us that the 429 might contain a `retry-after` value in the HTTP header[^1], are we sure that we don't want to attempt to fish that out of the headers?

[^1]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429

Reply:
Ah right, I had trouble with this for the QR login as well. We probably should do something about this, one of these days... :smiling_face_with_tear: